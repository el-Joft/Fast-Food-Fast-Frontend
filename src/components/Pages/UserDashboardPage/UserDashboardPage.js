/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/forbid-prop-types,react/require-default-props */
import React, { Component } from 'react';
import './UserDashboardPage.scss';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getOrders } from '../../../actions/orderAction';

export class UserDashboardPage extends Component {
  componentWillMount() {
    const { isAuthenticated } = this.props.auth;
    if (isAuthenticated) {
      const token = localStorage.getItem('userToken');
      const { id } = this.props.auth.user;
      this.props.getOrders(token, id);
    } else {
      this.props.history.push('/login');
    }
  }


  renderCard = () => this.props.orders.map(item => (


    <div key={item.id} className="history-body">

      <div className="history-item item">

        <h3>Item</h3>
        <div>
          <h2>{item.name}</h2>
          <p>
            {item.description}
          </p>
        </div>

      </div>
      <div className="history-item history-quantity">
        <h3>Quantity</h3>
        <p>{item.quantity}</p>
      </div>
      <div className="history-item history-price">
        <h3>Price</h3>
        <p>
          <strike>N</strike>
          {item.price}
        </p>
      </div>
      <div className="history-item history-total">
        <h3>Total</h3>
        <p>
          <strike>N</strike>
          {' '}
          {item.totalprice}
        </p>
      </div>
    </div>

  ));

  render() {
    return (
      <div>
        {
          this.props.orders === null
            ? null
            : (
              <div className="history">
                <div className="history-head-text">
                  <h3>YOUR ORDER HISTORY</h3>
                </div>
                {this.renderCard()}
              </div>
            )
        }
      </div>
    );
  }
}

UserDashboardPage.propTypes = {
  getOrders: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  orders: PropTypes.object,
  isAuthenticated: PropTypes.bool.isRequired,
  history: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
  orders: state.orders.orders,
});

export default connect(mapStateToProps, { getOrders })(UserDashboardPage);
