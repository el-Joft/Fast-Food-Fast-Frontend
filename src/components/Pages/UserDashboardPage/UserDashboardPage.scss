/* HISTORY */
.history{
    margin-bottom: 50px;
}
.history-head-text{
    text-align: center;
    font-size: 1.7em;
    padding: 13px;
    border: 2px solid #42ce9f;
    width: 40%;
    margin-top: 14px;
}
.history-body{
    display: flex;
    width: 70%;
    border-bottom: 2px solid #42ce9f;
    border-top: 2px solid #42ce9f;
    margin-top: 20px; 
}
.history-item{
    padding: 10px;
}
.history-item h3{
    background: #42ce9f;
    text-align: center;
    color: #fff;
    padding: 10px;
}
.history-item p{
    margin-top: 10px;
}
.item{
    flex-grow: 1;
    width: 40%;
}
.history-quantity, .history-date,
.history-price, .history-time, .history-total
{
    text-align: center;
}