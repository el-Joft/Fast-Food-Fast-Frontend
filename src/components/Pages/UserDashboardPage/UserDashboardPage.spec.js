/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { UserDashboardPage } from './UserDashboardPage';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      getOrders: jest.fn,
      orders: [],
      history: {
        push: jest.fn,
      },
      auth: {
        isAuthenticated: false,
        user: {
          id: 1,
        },
      },
    };
  });

  it('invokes `componentWillMount` when mounted', () => {
    const spy = jest.spyOn(UserDashboardPage.prototype, 'componentWillMount');
    wrapper = shallow(<UserDashboardPage {...props} />);
    wrapper.instance().componentWillMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should render the component', () => {
    wrapper = shallow(<UserDashboardPage {...props} />);
    expect(wrapper.length).toBe(1);
  });

  it('should render the component', () => {
    wrapper = shallow(<UserDashboardPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
