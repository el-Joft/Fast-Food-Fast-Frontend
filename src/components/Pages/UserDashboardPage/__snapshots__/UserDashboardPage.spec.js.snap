// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDashboardPage
    auth={
      Object {
        "isAuthenticated": false,
        "user": Object {
          "id": 1,
        },
      }
    }
    getOrders={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    orders={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="history"
      >
        <div
          className="history-head-text"
        >
          <h3>
            YOUR ORDER HISTORY
          </h3>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="history-head-text"
          >
            <h3>
              YOUR ORDER HISTORY
            </h3>
          </div>,
          Array [],
        ],
        "className": "history",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              YOUR ORDER HISTORY
            </h3>,
            "className": "history-head-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "YOUR ORDER HISTORY",
            },
            "ref": null,
            "rendered": "YOUR ORDER HISTORY",
            "type": "h3",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="history"
        >
          <div
            className="history-head-text"
          >
            <h3>
              YOUR ORDER HISTORY
            </h3>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="history-head-text"
            >
              <h3>
                YOUR ORDER HISTORY
              </h3>
            </div>,
            Array [],
          ],
          "className": "history",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h3>
                YOUR ORDER HISTORY
              </h3>,
              "className": "history-head-text",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "YOUR ORDER HISTORY",
              },
              "ref": null,
              "rendered": "YOUR ORDER HISTORY",
              "type": "h3",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
